app/services/set_decline_by_default.rb

```diff
@@ -0,0 +1,38 @@
+class SetDeclineByDefault
```

Comment:
Once all applications are rejected/offered/withdrawn, we also want to email the candidate - otherwise they won't know that an offer has been made. Would be good to structure the code so that adding that will be easy (renaming `SetDeclineByDefault` should work, or separating out the code that checks if we need to set decline by default from the actual setting). 

In the near future we also need to detect whether or not all of the applications are rejected, so that the candidate can be put into "Apply 2" - worth keeping in mind too.

Reply:
The pending check will now be based on ```status == 'awaiting_provider_decision'```, so perhaps we could have a helper method on ApplicationForm, e.g. ```#awaiting_provider_decisions?```

This could be used by SetDeclineByDefault, but also from PutCandidateIntoApply2 (with an additional check there are no offers).